### ISSUE TITLE
iOS Build Failed - ":CFBundleIdentifier", Does Not Exist

### ISSUE THREAD

**Issue by Wesley-Brooks** [invertase/react-native-firebase-starter#33]
I'm trying to run the base starter project. I have followed the readme instructions step by step twice, but I get the following error when running `npm run ios`. My npm/node versions are: 5.6.0/10.0.0. I can provide more info if necessary. 

```
** BUILD FAILED **


The following build commands failed:

	CompileC /Users/wesleybrooks/SideProjects/ng2Bondfire/react-native-firebase-starter/ios/build/Build/Intermediates/React.build/Debug-iphonesimulator/React.build/Objects-normal/x86_64/RCTShadowView+Internal.o Views/RCTShadowView+Internal.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler
(1 failure)

Installing build/Build/Products/Debug-iphonesimulator/Firescreen.app
An error was encountered processing the command (domain=NSPOSIXErrorDomain, code=2):
Failed to install the requested application
An application bundle was not found at the provided path.
Provide a valid path to the desired application bundle.
Print: Entry, ":CFBundleIdentifier", Does Not Exist

Command failed: /usr/libexec/PlistBuddy -c Print:CFBundleIdentifier build/Build/Products/Debug-iphonesimulator/Firescreen.app/Info.plist
Print: Entry, ":CFBundleIdentifier", Does Not Exist


npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! Firescreen@0.1.0 ios: `react-native run-ios`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the Firescreen@0.1.0 ios script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/wesleybrooks/.npm/_logs/2018-05-13T01_41_34_792Z-debug.log
```

**Comment by Salakar** [comment]
Could you show the contents of your iOS's `Info.plist` file?

**Comment by Wesley-Brooks** [comment]
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>LastAccessedDate</key>
	<date>2018-05-13T01:40:05Z</date>
	<key>WorkspacePath</key>
	<string>/Users/wesleybrooks/SideProjects/ng2Bondfire/react-native-firebase-starter/ios/Firescreen.xcworkspace</string>
</dict>
</plist>
```

**Comment by Wesley-Brooks** [comment]
This is probably the Info.plist you were asking for.  At ios/<app-name>/Info.plist
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>CFBundleDevelopmentRegion</key>
	<string>en</string>
	<key>CFBundleDisplayName</key>
	<string>$(DISPLAY_NAME)</string>
	<key>CFBundleExecutable</key>
	<string>$(EXECUTABLE_NAME)</string>
	<key>CFBundleIdentifier</key>
	<string>$(PRODUCT_BUNDLE_IDENTIFIER)</string>
	<key>CFBundleInfoDictionaryVersion</key>
	<string>6.0</string>
	<key>CFBundleName</key>
	<string>$(DISPLAY_NAME)</string>
	<key>CFBundlePackageType</key>
	<string>APPL</string>
	<key>CFBundleShortVersionString</key>
	<string>1.0</string>
	<key>CFBundleSignature</key>
	<string>????</string>
	<key>CFBundleVersion</key>
	<string>1</string>
	<key>LSRequiresIPhoneOS</key>
	<true/>
	<key>UILaunchStoryboardName</key>
	<string>LaunchScreen</string>
	<key>UIRequiredDeviceCapabilities</key>
	<array>
		<string>armv7</string>
	</array>
	<key>UISupportedInterfaceOrientations</key>
	<array>
		<string>UIInterfaceOrientationPortrait</string>
		<string>UIInterfaceOrientationLandscapeLeft</string>
		<string>UIInterfaceOrientationLandscapeRight</string>
	</array>
	<key>UIViewControllerBasedStatusBarAppearance</key>
	<false/>
	<key>NSLocationWhenInUseUsageDescription</key>
	<string></string>
	<key>NSAppTransportSecurity</key>
	<!--See http://ste.vn/2015/06/10/configuring-app-transport-security-ios-9-osx-10-11/ -->
	<dict>
		<key>NSExceptionDomains</key>
		<dict>
			<key>localhost</key>
			<dict>
				<key>NSExceptionAllowsInsecureHTTPLoads</key>
				<true/>
			</dict>
		</dict>
	</dict>
</dict>
</plist>
```

**Comment by Wesley-Brooks** [comment]
Had to upgrade to the latest version of Xcode.

**Comment by TheHollidayInn** [comment]
I'm getting this issue with Xcode Version 9.3.1

**Comment by Wesley-Brooks** [comment]
My npm/node versions are 6.0.1/10.0.0, macOS 10.13.4, and Xcode 9.3.1 . From what I've noticed it's usually due to having the wrong version of something, also make sure you have no special characters in your app name.

**Comment by TheHollidayInn** [comment]
I just updated everything to the same as yours. The apps name is `TestingLite`. Not sure if uppercase is bad? I'm still getting the same error.

**Comment by scwicker** [comment]
I had the same issue and fixed it with the following change in the RCTJavaScriptLoader.mm file. 
Change "if (_total > 0)"  to "if ([_total intValue] > 0) "  It's towards the top of the file in the first set of functions. 

Take the following path from your node_modules folder to find the file. 
/node_modules/react-native/React/Base

My error originally said CFBundleIdentifier, but once I added the square brackets and intValue everything worked flawlessly.

**Comment by glockjt** [comment]
Just got this same issue. Xcode is up to date as well.

**Comment by DatMoser** [comment]
Also same problem here. Experiencing the same problem with the firebase starter kit. Support says it has to do with wrong placement of the GoogleService-File, but if you follow the instructions step by step, you still get this issue!

**Comment by DatMoser** [comment]
> I had the same issue and fixed it with the following change in the RCTJavaScriptLoader.mm file.
> Change "if (_total > 0)" to "if ([_total intValue] > 0) " It's towards the top of the file in the first set of functions.
> 
> Take the following path from your node_modules folder to find the file.
> /node_modules/react-native/React/Base
> 
> My error originally said CFBundleIdentifier, but once I added the square brackets and intValue everything worked flawlessly.

They already changed it in the current version, but the error still occurres :(

**Comment by scwicker** [comment]
> > I had the same issue and fixed it with the following change in the RCTJavaScriptLoader.mm file.
> > Change "if (_total > 0)" to "if ([_total intValue] > 0) " It's towards the top of the file in the first set of functions.
> > Take the following path from your node_modules folder to find the file.
> > /node_modules/react-native/React/Base
> > My error originally said CFBundleIdentifier, but once I added the square brackets and intValue everything worked flawlessly.
> 
> They already changed it in the current version, but the error still occurres :(

@DatMoser ,  Try my solution posted above and let me know if it works for you.

**Comment by scwicker** [comment]
Oh sorry, I misread that. Bummer, and weird that this issue keeps finding a way to resurface.

**Comment by glockjt** [comment]
Any resolution yet for this?  Still can't build for iOS.

**Comment by michaelcavallaro03** [comment]
@glockjt 
I found these slightly modified setup instructions to finally get this working for me. 

git clone https://github.com/invertase/react-native-firebase-starter
npm install
npm run rename
cd ios
pod repo update
pod install
cd ..
Add Google-Info.plist to ios/
Add google-services.json to android/app
npm run ios

**Comment by shaynos** [comment]
@michaelcavallaro03 

Thanks for your modified instructions, I have tried both your instructions and the instructions in the readme with no success, can I ask, what version of Xcode are you running?

I am running Xcode 9.4.1, the only reason I ask is that I've seen others post the same issue on the react-native repo, and the response there was that react-native is not yet compatible with Xcode 10.

**Comment by Salakar** [comment]
Make sure you're adding the Google services file through Xcode and not just by copying the file into the directory, needs to be done inside Xcode otherwise it won't link to the project.

**Comment by glockjt** [comment]
So here is what fixed it more for me:

1. `node_modules/react-native/scripts`
	a. run `./ios-install-third-party.sh`
	b run `./ios-configure-glog.sh`
2. `node_modules/react-native/third-party/glog-0.3.4`
	a. run `./configure`
	b. run `make`
	c. run `make install`

**Comment by michaelcavallaro03** [comment]
@shaynos I am running Xcode 10 on macOS Mojave. I also (found this out later on), that I needed to downgrade my react-native from 0.57.3 to 0.57.0. Building and Deploying without issues now.

**Comment by michaelcavallaro03** [comment]
> So here is what fixed it more for me:
> 
> 1. `node_modules/react-native/scripts`
>    a. run `./ios-install-third-party.sh`
>    b run `./ios-configure-glog.sh`
> 2. `node_modules/react-native/third-party/glog-0.3.4`
>    a. run `./configure`
>    b. run `make`
>    c. run `make install`

@glockjt This also worked for me for the record, I just wanted a solution that avoided modifying node_modules

**Comment by glockjt** [comment]
@michaelcavallaro03 I agree this is a fundamental error but a work around for now and hopefully not a long term solution.

**Comment by Salakar** [comment]
@glockjt @michaelcavallaro03 do those steps still allow you to archive on Xcode? Aren't you doubling the `configure` of glog?

Also RN57 is glog version `0.3.5` - best using that one.

**Comment by glockjt** [comment]
@Salakar at the time I installed `react-native-firebase`,  `react-native: "0.55.3"` was a part of the installation which is using `glog: "0.3.4"`.  Also either copying my google settings through the terminal or in Xcode didn't matter, I still had to do the above steps I listed.

**Comment by michaelcavallaro03** [comment]
@Salakar I had no issues, build/deploy or archiving, after running @glockjt steps. I also had no issues after a clean clone of react-native-firebase-starter and running my above steps.

**Comment by shaynos** [comment]
As it turns out, the `CFBundleIdentifier` error is actually a generic, non-specific error, which is actually leading people on a wild goose chase. Check out this issue on the react-native repo, it's explained a bit better.

https://github.com/facebook/react-native/issues/21303

This is not really react-native-firebase related, but I decided to try and build an empty react-native project:

Created a new project:
`react-native init MyApp`
`cd MyApp`
`react-native run-ios`

I ended up with the `CFBundleIdentifier` when trying to build, but if I looked back about 80 lines I also found:

`error: Build input file cannot be found: '/Users/Me/MyApp/node_modules/react-native/third-party/double-conversion-1.1.6/src/fast-dtoa.cc'`

So I did a bit of a search and found this on stackoverflow:

https://stackoverflow.com/questions/52401856/problems-after-upgrading-to-xcode-10-build-input-file-cannot-be-found

I opened the project in Xcode, went to file > Workspace Setting, and changed build setting to Legacy Build System.

I can't guarantee that this will work for anyone else, and I can't guarantee that using Legacy Build System will not give you a head ache in the future, but, it got me up and running for now.

I'll try and do another build with the react-native-firebase quickstart and see what I can find.
